File: worker.go
// worker.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "time"
)

func main() {
    fmt.Println("Go worker ready")
    for i := 0; i < 5; i++ {
        time.Sleep(1 * time.Second)
        fmt.Printf("Go says %d\n", i)
    }

    scanner := bufio.NewScanner(os.Stdin)
    for scanner.Scan() {
        msg := scanner.Text()
        fmt.Printf("Go got: %s\n", msg)
    }
}

File: worker.py
# worker.py
import sys
import time

print("Python worker ready", flush=True)

for i in range(5):
    time.sleep(1)
    print(f"Python says {i}", flush=True)

for line in sys.stdin:
    msg = line.strip()
    print(f"Python got: {msg}", flush=True)
